x/multi-staking/keeper/msg_server.go

```diff
@@ -50,33 +52,32 @@ func NewMsgServerImpl(keeper Keeper) *msgServer {
 }
 
 // UpdateParams updates the staking params
-func (k msgServer) UpdateParams(goCtx context.Context, msg *stakingtypes.MsgUpdateParams) (*stakingtypes.MsgUpdateParamsResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
+func (k msgServer) UpdateParams(ctx context.Context, msg *stakingtypes.MsgUpdateParams) (*stakingtypes.MsgUpdateParamsResponse, error) {
+	sdkctx := sdk.UnwrapSDKContext(ctx)
 
 	if k.keeper.authority != msg.Authority {
 		return nil, fmt.Errorf("invalid authority; expected %s, got %s", k.keeper.authority, msg.Authority)
 	}
 
 	// store params
-	if err := k.keeper.stakingKeeper.SetParams(ctx, msg.Params); err != nil {
+	if err := k.keeper.stakingKeeper.SetParams(sdkctx, msg.Params); err != nil {
 		return nil, err
 	}
 
 	return &stakingtypes.MsgUpdateParamsResponse{}, nil
 }
 
 // CreateValidator defines a method for creating a new validator
-func (k msgServer) CreateValidator(goCtx context.Context, msg *stakingtypes.MsgCreateValidator) (*stakingtypes.MsgCreateValidatorResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
-
-	multiStakerAddr, valAcc, err := types.AccAddrAndValAddrFromStrings(msg.DelegatorAddress, msg.ValidatorAddress)
+func (k msgServer) CreateValidator(ctx context.Context, msg *stakingtypes.MsgCreateValidator) (*stakingtypes.MsgCreateValidatorResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	multiStakerAddr, valAcc, err := types.ValidatorAccAddrAndValAddrFromStrings(msg.ValidatorAddress)
```

Comment:
This is false, validator address is not multi staker address, validator and delegator have different address' prefix and we don't need tto create new function to generate validator address.

Reply:
I've re-read sdk50, it still work, but I think we should add address codec and validator codec for transparent this logic